### PR TITLE
Make core module use old API (archived to clients/python_v1)

### PR DESCRIPTION
**Pull request by kbab**
Make core module use old API (archived to clients/python_v1)

### CODE REVIEW

File: tools/dev/pyproject.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ typer = {extras = ["all"], version = "^0.7.0"}
 fs = "^2.4.16"
 dev_bia-integrator-core = { path = "../../core/dev", develop = true }
 bia-integrator-api = { path = "../../clients/python", develop = true }
+pydantic = "2.4.0"
```
Comment (ctr26) [inline tools/dev/pyproject.toml:18]: Is this needed if bia-integrator-api depends on pydantic already?

Comment (ctr26) [reply tools/dev/pyproject.toml:18]: My concern would be if `bia-integrator-api `updates pydantic in the future then this pinned version will clash.

**Comment by ctr26** [comment]
These commits have a distinct sense of having the .pre-commit hook running. Is it working ok?

**Comment by kbab** [comment]
> These commits have a distinct sense of having the .pre-commit hook running. Is it working ok?

In my opinion - yes. It also does some useful things apart from formatting, like point out unused variables.

**Comment by ctr26** [comment]
Amazing! Glad it's working well for you, the linting stuff is really helpful for me too

**Review by liviuba — APPROVED** [review]
LGTM! Just to check, this is meant to just use the previous version of the api, not pass all tests yet, right?

**Pull request merged**